test/test_http11.py

```diff
@@ -511,6 +512,13 @@ def read(self, size):
         assert 'File-like bodies must return bytestrings. ' \
                'Got: {}'.format(int) in str(exc_info)
 
+    def test_exception_raised_on_socket_timeout(self):
+        c = HTTP11Connection('httpbin.org', socket_timeout=0.01)
```

Comment:
Thank you for taking the time to make these comments. I will adjust the changes above. For this line comment, the URL I am requesting is designed to block for n seconds

 c.request('GET', '/delay/1')

From the docs on httpbin "Delays responding for min(n, 10) seconds."

so I think that would save us from even Moore's Law. Please advise if I'm overlooking something.